features/steps/steps.py

```diff
@@ -143,7 +144,23 @@ def when_i_attach_staging_token(context, token_type, user_spec):
             ),
             user_spec,
         )
-    when_i_run_command(context, "ua attach %s" % token, user_spec)
+    cmd = "ua attach {}".format(token)
+    when_i_run_command(context, cmd, user_spec, verify_return=False)
+
+    retries = [5, 5, 10]  # Sleep times to wait between retries
+    while context.process.returncode != 0:
+        when_i_run_command(context, "ua detach --assume-yes", user_spec)
```

Comment:
I like the retry logic on inability to attach to staging (reasonable for our integration testing against staging because  that service goes down for deployment fairly regularly).

but I do think we should not have to call `ua detach` during those failures. UA  client should be smart enough on failure to attach to clean up any artifacts it created during attach failure. This is reminiscent of #1337 which I will put up a fix for shortly and we should land that before this PR so that you can drop the `ua detach` I hope.
```suggestion
```

Reply:
Okay, fair enough I did that because when I was saw that issue locally, I saw that the machine kept attached. But I could be wrong on that. 

I will drop this line and keep an eye on CI to see if it surface some errors like that